storagedriver/s3/s3.go

```diff
@@ -642,6 +642,15 @@ func (d *Driver) URLFor(path string, options map[string]interface{}) (string, er
 		return "", storagedriver.InvalidPathError{Path: path}
 	}
 
+	methodString := "GET"
+	method, ok := options["method"]
+	if ok {
+		if method != "GET" && method != "HEAD" {
+			return "", storagedriver.ErrUnsupportedMethod
+		}
+		methodString, _ = method.(string)
```

Comment:
Exactly. That statement continuing to not crash relies on a side-effect of the if statement above. If this condition gets changed in any way such that the guarantee is removed, it will crash, unexpectedly.